falcon/cyutil/uri.pyx

```diff
@@ -31,19 +31,33 @@ cdef void _init_hex_table(int* data):
                 pass
 
 
+cdef void _init_hex_encoded_table(int* data):
+    cdef Py_ssize_t index
+    cdef unsigned char b1
+    cdef unsigned char b2
+
+    for index in range(0x100):
+        b1, b2 = f'{index:02X}'.encode()
+        data[index] = (b1 << 8) + b2
```

Comment:
don't we need to cast to int at least one of the two? or `b1 << 8` does automatically cast a char (1byte) to short(2 bytes)/int(4bytes)?

Reply:
No, this function has no impact on perf at all, because it is used to prepare a lookup table in advance.

I see what you mean, no, it shouldn't be a problem, these kind of expressions are normally computed at least as 32 bit. But I can change the types just to stay on the safe side, yes!

Reply:
it's likely just me that's rusty in c semantics. given that everything works it's likely as you say

Reply:
Yeah, but things like that may vary depending on the compiler, OS, and architecture, so I like the idea of fixing it properly.